lib/webpush/request.rb

```diff
@@ -36,17 +44,26 @@ def headers
       headers["Content-Type"] = "application/octet-stream"
       headers["Ttl"]          = ttl
 
-      if encrypted_payload?
+      if @payload.has_key?(:server_public_key)
         headers["Content-Encoding"] = "aesgcm"
-        headers["Encryption"] = "salt=#{salt_param}"
-        headers["Crypto-Key"] = "dh=#{dh_param}"
+        headers["Encryption"] = "keyid=p256dh;salt=#{salt_param}"
+        headers["Crypto-Key"] = "keyid=p256dh;dh=#{dh_param}"
       end
 
-      headers["Authorization"] = "key=#{api_key}" if api_key?
+      vapid_headers = build_vapid_headers
+      headers["Authorization"] = vapid_headers["Authorization"]
```

Comment:
I think this still needs the previous case matching statement, or something to the effect of:
```ruby
if (@endpoint =~ /\Ahttps:\/\/(android|gcm-http)\.googleapis\.com/) {
  vapid_headers = api_key? ? {"Authorization": "key=#{api_key}" } : {}
} else {
  vapid_headers = build_vapid_headers
}
```

Reason being that for backwards compatibility, people may still have old tokens saved. Basically iff the subscription is created with a VAPID public key in JS on the client, an FCM endpoint will be returned. Otherwise, you still get the old style and wouldn't be able to perform VAPID with it.